health/service/src/main/java/com/cerner/beadledom/health/HttpHealthDependency.java

```diff
@@ -0,0 +1,54 @@
+package com.cerner.beadledom.health;
+
+import java.io.IOException;
+import java.util.Objects;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.Response;
+
+/**
+ * Implementation of {@link HealthDependency} to check the health of any HTTP service.
+ *
+ * @author Venkatesh Sridharan.
+ */
+public class HttpHealthDependency extends HealthDependency {
+  private final String availabilityUrl;
+  private final String name;
+  private final OkHttpClient client;
+
+  /**
+   * Constructs an instance of {@link HttpHealthDependency} with the given parameters.
+   *
+   * @param availabilityUrl String representing URL of the availability health check.
+   * @param name String representing name of the dependency
+   * @param isPrimary Boolean representing whether the dependency is primary or not.
+   */
+  public HttpHealthDependency(String availabilityUrl, String name, boolean isPrimary) {
+    super(isPrimary);
+    Objects.requireNonNull(availabilityUrl);
+    Objects.requireNonNull(name);
+
+    this.availabilityUrl = availabilityUrl;
+    this.name = name;
+    this.client = new OkHttpClient.Builder().build();
```

Comment:
From the OkHttpClient docs,

```
OkHttp performs best when you create a single `OkHttpClient` instance and reuse it for all of your HTTP calls.
```

Lets update the constructor to take in a OkHttpClient. Doing this would also allow consumer to configure additional interceptors.

Reply:
I wonder if something like this is an exclusion to that. We aren't really concerned about performance here and are just choosing OkHttp as the implementation of a simple health check call.

Reply:
Here is some addittional information, https://square.github.io/okhttp/4.x/okhttp/okhttp3/-ok-http-client/#okhttpclients-should-be-shared 

> This is because each client holds its own connection pool and thread pools. Reusing connections and threads reduces latency and saves memory.

We could have two constructors, one the has the OkHttpClient provided, and one the creates one.

Reply:
But shouldn't each Health dependency have its own client instance created though? Because then we would just be initializing the class once for each dependency. I also didn't want to expose that we are using OkHttp

Reply:
@nathanschile While I do think the okhttp docs are recommending a single client instance being shared, I think their biggest recommendation is against creating a new client for every request. 

Is reusing a client instance per dependency but not across the whole service worth the tradeoff for the value we get in not exposing the implementation (namely that we can change implementations and makes it simpler to use)?

Reply:
I think it is fine to have it as an implementation detail, thus not exposing it. If we determine at a later date that it is a issue having a client per dependency, then we have an option to expose it at that time.